File: Backend/src/Agro_Trade.Domain/Entities/ImpactoSocial.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace Agro_Trade.Domain.Entities
{
    public class ImpactoSocial
    {
        [Key]
        public int IdImpacto { get; set; }
        public int IdProveedor { get; set; }
        public Proveedor Proveedor { get; set; } = null!;
        public int IdPedido { get; set; }
        public Pedido Pedido { get; set; } = null!;
        public int IdDetallePedido { get; set; }
        public DetallePedido DetallePedido { get; set; } = null!;

        public DateTime FechaRegistro { get; set; } = DateTime.UtcNow;
        public float ProductosSalvados { get; set; }
        public float BeneficioExtraProductor { get; set; }
    }
}

File: Frontend/agrotrade_frontend/lib/models/Productor/metricas/impacto_social.dart
class ImpactoSocial {
  final int idImpacto;
  final int idProveedor;
  final int idPedido;
  final int idDetallePedido;
  final DateTime fechaRegistro;
  final double productosSalvados;
  final double beneficioExtraProductor;

  const ImpactoSocial({
    required this.idImpacto,
    required this.idProveedor,
    required this.idPedido,
    required this.idDetallePedido,
    required this.fechaRegistro,
    required this.productosSalvados,
    required this.beneficioExtraProductor,
  });

  factory ImpactoSocial.fromJson(Map<String, dynamic> json) {
    return ImpactoSocial(
      idImpacto: json['idImpacto'] as int,
      idProveedor: json['idProveedor'] as int,
      idPedido: json['idPedido'] as int,
      idDetallePedido: json['idDetallePedido'] as int,
      fechaRegistro: json['fechaRegistro'] != null
          ? DateTime.tryParse(json['fechaRegistro'] as String) ?? DateTime.now()
          : DateTime.now(),
      productosSalvados: (json['productosSalvados'] as num?)?.toDouble() ?? 0.0,
      beneficioExtraProductor: (json['beneficioExtraProductor'] as num?)?.toDouble() ?? 0.0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idImpacto': idImpacto,
      'idProveedor': idProveedor,
      'idPedido': idPedido,
      'idDetallePedido': idDetallePedido,
      'fechaRegistro': fechaRegistro.toIso8601String(),
      'productosSalvados': productosSalvados,
      'beneficioExtraProductor': beneficioExtraProductor,
    };
  }

  String get beneficioFormateado {
    return '\$${beneficioExtraProductor.toStringAsFixed(2)}';
  }
}
